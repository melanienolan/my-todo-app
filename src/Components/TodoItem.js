import React from 'react';
import PropTypes from 'prop-types';
import Check from 'react-icons/lib/fa/check';
import Close from 'react-icons/lib/fa/close';

import CircleThin from 'react-icons/lib/fa/circle-thin';

const TodoItem = props => {
  return (
    <div className="todo-list__item">
      <button
        className="todo-list__button--completed"
        onClick={() => props.onCompleted(props.todo)}
      >
        {props.todo.completed ? <Check /> : <CircleThin />}
      </button>
      <p className="todo-list__text">
        {props.todo.title}
      </p>
      <button
        className="todo-list__button--delete"
        onClick={() => props.onDeleted(props.todo)}
      >
        <Close />
      </button>
    </div>
  );
};

TodoItem.propTypes = {
  todo: PropTypes.object,
  onCompleted: PropTypes.func,
  onDeleted: PropTypes.func
};

export default TodoItem;
