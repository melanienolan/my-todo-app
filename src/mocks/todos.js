export const todos = [
  {
    completed: true,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'complete todo'
  },
  {
    completed: false,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'incomplete todo'
  }
];
