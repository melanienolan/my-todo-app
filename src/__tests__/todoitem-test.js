import React from 'react';
import TodoItem from '../Components/TodoItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { todos } from '../mocks/todos';

const completeTodo = todos[0];
const incompleteTodo = todos[1];

it('renders complete todo without crashing', () => {
  shallow(<TodoItem todo={completeTodo} />);
});

it('renders incomplete todo without crashing', () => {
  shallow(<TodoItem todo={incompleteTodo} />);
});

it('renders complete todo item', () => {
  const todoItemComponent = shallow(<TodoItem todo={completeTodo} />);
  expect(toJson(todoItemComponent)).toMatchSnapshot();
});

it('renders incomplete todo item', () => {
  const todoItemComponent = shallow(<TodoItem todo={incompleteTodo} />);
  expect(toJson(todoItemComponent)).toMatchSnapshot();
});
