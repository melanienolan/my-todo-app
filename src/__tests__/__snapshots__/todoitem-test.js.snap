// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders complete todo item 1`] = `
<div
  className="todo-list__item"
>
  <button
    className="todo-list__button--completed"
    onClick={[Function]}
  >
    <FaCheck />
  </button>
  <p
    className="todo-list__text"
  >
    complete todo
  </p>
  <button
    className="todo-list__button--delete"
    onClick={[Function]}
  >
    <FaClose />
  </button>
</div>
`;

exports[`renders incomplete todo item 1`] = `
<div
  className="todo-list__item"
>
  <button
    className="todo-list__button--completed"
    onClick={[Function]}
  >
    <FaCircleThin />
  </button>
  <p
    className="todo-list__text"
  >
    incomplete todo
  </p>
  <button
    className="todo-list__button--delete"
    onClick={[Function]}
  >
    <FaClose />
  </button>
</div>
`;
